### PR TITLE
Typography changes

### PR DESCRIPTION
**Pull request by ClementChaumel**
## Done

[List of work items including drive-bys]

Fixes [list issues/bugs if needed]

## QA

- Open [demo](insert-demo-url)
- [Add QA steps]
- Review updated documentation:
  - [List any updated documentation for review]

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [ ] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

## Screenshots

[if relevant, include a screenshot or screen capture]

### CODE REVIEW

**Comment by bartaz** [comment]
@ClementChaumel @lyubomir-popov Percy results for review and reference: https://percy.io/bb49709b/vanilla-framework/builds/26624752/changed/1482189495

File: scss/_settings_spacing.scss [hunk 1]
```diff
@@ -8,44 +8,44 @@ $sp-unit-ratio: 0.5 !default;
...
-  h4-mobile: 0.3rem,
+  h1-large: 0.5rem,
+  h1-mobile: 0.5rem,
+  h1: 0.5rem,
+  h2: 0.5rem,
+  h2-mobile: 0.5rem,
+  h3: 0.45rem,
+  h3-mobile: 0.45rem,
+  h4-large: 0.45rem,
+  h4: 0.45rem,
```
Comment (bartaz) [inline scss/_settings_spacing.scss:47]: @lyubomir-popov This change is causing a huge shift. From almost nothing, to 0.45rem it's pushing all h4s (and related headings in components) almost whole spacing unit lower.

How should we compensate for that? Move everything else lower as well (icons that should align with that?

Comment (lyubomir-popov) [reply scss/_settings_spacing.scss:47]: Yes, the good thing is that all nudges are roughly the same value - so padding on top of text becomes more consistent.

**Comment by ClementChaumel** [comment]
Should this be closed in favour of the 4.0 branch?

**Comment by bartaz** [comment]
This can be closed as we have an official `vanilla-4.0` branch, demo available in #4751

**Pull request closed without merging**